.buildkite/steps/script.sh

```diff
@@ -18,6 +18,12 @@ echo "+++ running tests"
 exitCode=$?
 py.test -ra --cov=stellargraph tests/ --doctest-modules --doctest-modules --cov-report=term-missing -p no:cacheprovider --junitxml="./${junit_file}" || exitCode=$?
 
+if [ "${CHECK_NOTEBOOK_FORMATTING-0}" = 1 ]; then
+  echo "+++ checking notebook formatting"
+  # This script requires non-trivial dependencies, so run it here after installing them
+  python scripts/format_notebooks.py --all --ci demos/ || exitCode=$?
```

Comment:
Could this be separated in a different parallel step?

Reply:
The comment above is meant to describe why we aren't doing this, but I can see that it's unclear.

Basically, we'd have to spend 2 minutes of CPU time installing dependencies for this step that takes 20s, which seems like a waste.

Reply:
I've updated the comment.